typewriter/fixes/fixer_utils.py

```diff
@@ -105,6 +211,118 @@ def is_import_stmt(node):
     root.insert_child(insert_pos, Node(syms.simple_stmt, children))
 
 
+# -------------- TYPE_CHECKING LOGIC ------------------ #
+
+
+def _new_type_check_with_import(package, name, root, insert_pos):
+    # type: (Optional[str], str, Node, int) -> None
+    """
+    Inserts a new TYPE_CHECKING block containing a new import statement for package and name
+
+    Parameters
+    -----------
+    package : Optional[str]
+    name : str
+    root : Node
+    insert_pos : int
+    """
+    # [Grammar]
+    # if_stmt: 'if' namedexpr_test ':' suite ('elif' namedexpr_test ':' suite)* ['else' ':' suite]
+    type_check_node = Node(syms.if_stmt,
+                           [Leaf(1, 'if'),
+                            Node(syms.power, [Leaf(1, 'typing'),
+                                              Node(syms.trailer, [Leaf(23, '.'),
+                                                                  Leaf(1, 'TYPE_CHECKING')])],
+                                 prefix=" "),
+                            Leaf(11, ':'),
+                            # [Grammar]
+                            # suite: simple_stmt | NEWLINE INDENT stmt+ DEDENT
+                            Node(syms.suite, [Leaf(token.NEWLINE, '\n'),
+                                              Leaf(token.INDENT, '    '),
+                                              Leaf(token.DEDENT, '')])])
+
+    children = [_generate_import_node(package, name), Newline()]
+    # We can just hardcode the correct insert position since we just created the typing block
+    type_check_node.children[3].insert_child(2, Node(syms.simple_stmt, children))
```

Comment:
why do this insert at all? When can just insert it directly up above, when we make the `syms.suite`.

Or... even better... make this function return a completely empty if-block, ie:

```
def _new_if_type_checking_suite(root, insert_pos):
```

...and then we don't have to repeat our code for creating / inserting the import statement, in `create_type_checking_import`

Reply:
Re just returning the empty if-block: I think we have the same amount of code either way. One import statement (the one immediately following the `if` line) has a different prefix than the others (which is a stupid CST design decision, in my opinion).  I will just add the code directly to the suite in the new type_checking node per your first suggestion